crates/neon/src/types_impl/date.rs

```diff
@@ -13,7 +13,45 @@ use crate::{
     sys::{self, raw},
 };
 
-/// A JavaScript Date object
+/// The type of JavaScript
+/// [`Date`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+/// objects.
+///
+/// # Example
+///
+/// The following shows an example of converting Rust
+/// [`SystemTime`](std::time::SystemTime) timestamps to JavaScript `Date` objects.
+///
+/// ```
+/// # use neon::prelude::*;
+/// use easy_cast::Cast; // for safe numeric conversions
+/// use neon::types::JsDate;
+/// use std::fs::File;
+/// use std::time::SystemTime;
+///
+/// /// Return the "modified" timestamp for the file at the given path.
+/// fn last_modified(path: &str) -> Option<f64> {
+///     // A more robust implementation would do more detailed error reporting.
+///     File::open(&path).ok()?
+///         .metadata().ok()?
+///         .modified().ok()?
+///         .duration_since(SystemTime::UNIX_EPOCH).ok()?
+///         .as_millis().try_cast().ok()
+/// }
+///
+/// fn modified(mut cx: FunctionContext) -> JsResult<JsDate> {
+///     let path: Handle<JsString> = cx.argument(0)?;
+///
+///     let modified = last_modified(&path.value(&mut cx))
+///         .and_then(|n| cx.date(n).ok());
+///
+///     if let Some(date) = modified {
+///         Ok(date)
+///     } else {
+///         cx.throw_error("failed to get timestamp")
+///     }
+/// }
```

Comment:
A small suggestion for simplifying this is using `Result<f64, Box<dyn Error>>` since everything implements `std::error::Error`, including `DateError`.

```suggestion
/// use std::{error::Error, fs::File, time::SystemTime};
///
/// /// Return the "modified" timestamp for the file at the given path.
/// fn last_modified(path: &str) -> Result<f64, Box<dyn Error>> {
///     Ok(File::open(&path)?
///         .metadata()?
///         .modified()?
///         .duration_since(SystemTime::UNIX_EPOCH)?
///         .as_millis()
///         .try_cast()?)
/// }
///
/// fn modified(mut cx: FunctionContext) -> JsResult<JsDate> {
///     let path: Handle<JsString> = cx.argument(0)?;
///
///     last_modified(&path.value(&mut cx))
///         .and_then(|n| Ok(cx.date(n)?))
///         .or_else(|err| cx.throw_error(err.to_string()))
/// }
```